### ISSUE TITLE
I have been dev for 20 years and I can say with guarantee this is the WORST documentation I have seen in my life.

### ISSUE THREAD

**Issue by ganeshkrishnan1** [amzn/selling-partner-api-docs#180]
I can understand typos. I can understand logical errors of mixing AWS with MWS though they are not the same.
BUT 
missing code?

uncompilable code?

urls pointing to amazon.com instead of sellercentral.amazon.com?

Horrible documentation, horrible code. Team of 3 devs splitting our hairs over this piece of crappy documentation for weeks and making no headway.

Is there a way startups can get together and donate some money to make this documentation a bit readable and not give migraine every time I scroll by it?

**Comment by fengmiq** [comment]
@ganeshkrishnan1 Amazon lets you convert java，I have broken down by php

**Comment by brinternational** [comment]
Docs are horrible
Don't bother submitting a ticket or thinking an Amazon rep will reply to this thread. 

Amazon rep will read all threads, only reply to ones they can answer, all others they ignore.. 
Welcome to working with Amazon, hands down the worst customer service on the planet.

**Comment by hellcoder420** [comment]
flipkart is even worse......

**Comment by fernlop** [comment]
I can't even access tax related information or upload invoices so i cannot adhere to legal requirements but nobody seems interested. Opened 2 issues, 2 support cases but no help.

**Comment by JacksonJeans** [comment]
Whoa!!! And I thought I was the only one who felt that way. 😄  Thank you @ganeshkrishnan1  for writing this! 👍 

I would be immediately involved in cleaning up and learning something new - how not to do it. But unfortunately I cannot do this for free....  Would I think even the fewest would do it? I try to help as much as I can in the issues and my spare time allows me to do so.

**Comment by neil-119** [comment]
Amazon is poor, so everybody pitch in a dollar.

No, seriously, I was so excited when I learned that Amazon was creating a RESTful API. Finally, I thought, back to modernity!

NOPE.

This is literally just a dressed up MWS API. Come on, man, we don't want more feeds and reports. A real REST API would have e.g. GET /api/inventory or if has to be async then at least webhooks or server push or SOMETHING more modern. What is this "polling" thing you speak of in 2020? /s

The documentation could be made way simpler. It took me four hours to integrate this. All I wanted to do was get it working with my own seller account and it wasn't immediately apparent that I could skip half the developer guide. Also the C# library is in pre-pre-alpha stage. You read that right. That thing is missing a million things.

Dude, why do I have to create and then encrypt a TSV string in memory with Amazon-managed encryption keys (and then upload to S3), jeebus, why can't I just POST my request with a normal JSON schema like every other API in the world? God in what world is this called a REST API? I should be able to POST an array of simple objects saying this SKU/ASIN is this quantity etc, but nope, that would be too productive for the developer, we can't have that happening. You need to be at least level 9000 to use Amazon's APIs.

Sorry, just needed to rant with other fellow frustrated developers here. I feel better now. I am glad that I am over level 9000 and I sincerely feel the pain of all of the other developers who aren't. Ok, I'm done being a sarcastic *** now. Good night.

**Comment by fernlop** [comment]
Is there a way to use the old MWS API? At least there is some documentation and it works :D 
Amazon forces me to use the new SP API because i registered as a developer after they released it. So i don't have any MWS Access.

**Comment by JacksonJeans** [comment]
@fernlop There is no way. If you didn't have access before, you won't get it today. And yes,.. wrong thread for so nh question 😄

**Comment by BeeBee234** [comment]
Hear, hear and huzzah to my fellow suffering programmers!!! 

Did most of you think this was a 100% usable solution when you first started?  Well, you'd think it would be since it's coming from a US$1 TRILLION monopoly company with essentially unlimited resources and access to the best programmers the world has ever seen!  But you were wrong.

You were wrong and now you are the guinea pigs who are working for Amazon for free (that's right for nothing!) to debug the entire API and write all of the documentation and example code.  

If this thing was fully cooked and ready for dinner, you'd be all done with your SP API app by now and it would be in production.  If you'd have listened to your intuition you would have stuck with the old API and not even looked at this until Amazon forced you to in 3 years, just like you learned the hard way with Paypal, Google, and the other monster companies.

As it stands we are on our own with no one on Amazon's end answering posts or support tickets and with a few Amazon super-fans cheering us on to not trash talk the API or Amazon support.  This is bad business.

Amazon: Please hire the superheros who are doing all of the work for you on this forum or we will switch to your competitors' solution.  Oh, that's right!  You'd don't have any competition.

**Comment by bobolobo** [comment]
So you mean the hours of endless interviews, quizzes, and tests that Amazon (and other big hitech companies) put prospective job candidates through does NOTHING to actually hire quality people... why color me shocked.

**Comment by thelastinuit** [comment]
> I can understand typos. I can understand logical errors of mixing AWS with MWS though they are not the same.
> BUT
> missing code?
> 
> uncompilable code?
> 
> urls pointing to amazon.com instead of sellercentral.amazon.com?
> 
> Horrible documentation, horrible code. Team of 3 devs splitting our hairs over this piece of crappy documentation for weeks and making no headway.
> 
> Is there a way startups can get together and donate some money to make this documentation a bit readable and not give migraine every time I scroll by it?

I did a gig for a startup... I had to create a lot of workarounds to handle the Product API because THERE WAS NO WARRANTY that the response was consistent. Sometimes fields appear out of effing nowhere.
At some point I thought I was querying to a black hole.

**Comment by ronlobo** [comment]
Haha, same same, working w/ the Amazon APIs was one of the worst experiences I ever had in my life, go figure Jeff.

**Comment by seanevan** [comment]
Hi all,

I closed and locked this issue earlier today as the discussion was increasingly diverging from specific issues with the documentation. Our team has been reading the feedback here and on HN today and reflecting on it. We have allocated additional capacity to address issues/bugs in the documentation. To help us target our efforts we would appreciate specific reports of content that is inaccurate.

Thanks,
Sean Evans
Manager, Support Engineering
